app/highlights/components/newHighlightList.js

```diff
@@ -0,0 +1,25 @@
+const h = require('vdux/element').default
+
+const { map } = require('lodash')
+
+const actions = require('../actions')
+
+module.exports = {
+  render,
+  onCreate
+}
+
+function render ({ props, children }) {
+  return h('div', {id: 'newHighlightList'}, map(children, (child, childKey) => {
+    return h(child, {}, [])
+  }),
+  h('button', {onClick: addHighlight }, 'Add new highlight'))
+}
+function addHighlight() {
+  //How to add a component I don't know about?
```

Comment:
you could add a new highlight by dispatching a change in internal component state or a feathers create action. in both cases, the components should be written such that when the state changes, the component's render function displays the state appropriately. this is the concept of "unidirectional data flow": state flows into the render functions and actions change state which causes a re-flow.